test/configCases/code-generation/string-chunk-id/webpack.config.js

```diff
@@ -0,0 +1,50 @@
+function FixModuleIdAndChunkIdPlugin() {}
+module.exports = FixModuleIdAndChunkIdPlugin;
+FixModuleIdAndChunkIdPlugin.prototype.apply = function(compiler) {
+	var hashFunction = compiler.options.output.hashFunction;
+	var hashDigest = compiler.options.output.hashDigest;
+	var hashDigestLength = compiler.options.output.hashDigestLength;
+	compiler.plugin("compilation", function(compilation) {
+		compilation.plugin("before-module-ids", function(modules) {
+			modules.forEach(function(module) {
+				if(module.id === null) {
+					var hash = require("crypto").createHash(hashFunction);
+					if(!module.resource) {
+						return;
+					}
+					var nodeModulesPathIndex = module.resource.indexOf('node_modules');
+					//hash is based on module's relative pathname, which is alway unique and fixed between different bundles
+					if(nodeModulesPathIndex > 0) {
```

Comment:
this should probably be `nodeModulesPathIndex > -1` right? (even though > 0 will probably work anyways)